src/live/src/user-to-user/pubnub-listener.js

```diff
@@ -0,0 +1,56 @@
+import { EventEmitter } from 'events';
+
+/**
+ * @typedef PubNubMessage
+ * @property {string} channel The channel for which the message belongs
+ * @property {string} subscription The channel group or wildcard subscription match (if exists)
+ * @property {string|number} timetoken Publish timetoken
+ * @property {Object} message The Payload
+ */
+
+/**
+ * @typedef PubNubPresence
+ * @property {string} action Can be join, leave, state-change or timeout
+ * @property {string} channel The channel to which the message belongs
+ * @property {string|number} occupancy Number of users connected to the channel
+ * @property {string} state User State
+ * @property {string} subscription The channel group or wildcard subscription match (if exists)
+ * @property {string|number} timestamp Current timetoken
+ * @property {string|number} timetoken Publish timetoken
+ * @property {string[]} uuid UUIDs of users who are connected with the channel
+ */
+
+/**
+ * @typedef PubNubStatus
+ * @property {string} category The status category. An enum is in the PubNub docs. Example: PNConnectedCategory
+ * @property {string} operation The operation the status is for. Example: PNSubscribeOperation
+ * @property {string[]} affectedChannels The channels affected in the operation
+ * @property {string[]} subscribedChannels All the current subscribed channels
+ * @property {string[]} affectedChannelGroups The channel groups affected in the operation
+ * @property {number} lastTimetoken The last timetoken used in the subscribe request
+ * @property {number} currentTimetoken The current timetoken fetched in the subscribe response, which is going to be used in the next request
+ */
+
+export class PubNubListener extends EventEmitter {
+  static statusPrefix = 'status:';
```

Comment:
If this is a constant value then we should define it as a `const` outside of the class. By defining as a `static` variable on the class it allows the variable to possibly be changed outside of the class.

```javascript
const STATUS_PREFIX = 'status:';

export class PubNubListener extends EventEmitter {
  // ...
}
```